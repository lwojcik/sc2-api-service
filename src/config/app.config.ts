import { registerAs } from '@nestjs/config';

export const appConfig = registerAs('app', () => ({
  name: 'sc2-api-service',
  environment: process.env.NODE_ENV,
  host: process.env.SAS_APP_HOST,
  port: parseInt(process.env.SAS_APP_PORT, 10),
  enableCors: process.env.SAS_APP_CORS_ENABLE === 'true',
  corsOrigin: process.env.SAS_APP_CORS_ORIGIN,
}));
