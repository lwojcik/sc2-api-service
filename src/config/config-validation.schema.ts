import * as Joi from 'joi';
import { validateRegionName } from '../common/validators/validateRegionName.validator';

const appSchema = {
  NODE_ENV: Joi.string().default('production'),
  SAS_APP_HOST: Joi.string().default('0.0.0.0'),
  SAS_APP_PORT: Joi.string().default(3000),
  SAS_APP_CORS_ENABLE: Joi.string().default('false'),
  SAS_APP_CORS_ORIGIN: Joi.optional(),
};

const redisSchema = {
  SAS_REDIS_ENABLE: Joi.string().default('false'),
  SAS_REDIS_HOST: Joi.string().optional().default('redis'),
  SAS_REDIS_PORT: Joi.string().optional().default('6379'),
  SAS_REDIS_PASSWORD: Joi.any().optional(),
  SAS_REDIS_TTL_SECS: Joi.string().optional().default('2000'),
  SAS_REDIS_DB: Joi.string().optional().default('0'),
};

const battlenetSchema = {
  SAS_BATTLENET_REGION: Joi.string().required().custom(validateRegionName),
  SAS_BATTLENET_CLIENT_ID: Joi.string().required(),
  SAS_BATTLENET_CLIENT_SECRET: Joi.string().required(),
  SAS_BATTLENET_TIMEOUT_MS: Joi.string().default('30000'),
};

const throttleSchema = {
  SAS_THROTTLE_LIMIT: Joi.string().default('300'),
  SAS_THROTTLE_TTL_SECS: Joi.string().default('60'),
};

const authSchema = {
  SAS_AUTH_ENABLE: Joi.string().default('false'),
  SAS_AUTH_USERNAME: Joi.string().optional(),
  SAS_AUTH_JWT_SECRET: Joi.string().optional(),
};

const httpsSchema = {
  SAS_HTTPS_ENABLE: Joi.string().default('false'),
  SAS_HTTPS_KEY_PATH: Joi.string().optional(),
  SAS_HTTPS_CERT_PATH: Joi.string().optional(),
};

const basSchema = {
  SAS_BAS_URL: Joi.string().required(),
};

export const configValidationSchema = Joi.object({
  ...appSchema,
  ...redisSchema,
  ...battlenetSchema,
  ...throttleSchema,
  ...authSchema,
  ...httpsSchema,
  ...basSchema,
});
