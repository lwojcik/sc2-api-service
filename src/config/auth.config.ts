import { registerAs } from '@nestjs/config';

export const authConfig = registerAs('auth', () => ({
  enable: process.env.SAS_AUTH_ENABLE === 'true',
  username: process.env.SAS_AUTH_USERNAME,
  jwtSecret: process.env.SAS_AUTH_JWT_SECRET,
}));
