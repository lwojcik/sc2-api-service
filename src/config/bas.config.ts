import { registerAs } from '@nestjs/config';

export const basConfig = registerAs('bas', () => ({
  url: process.env.SAS_BAS_URL,
}));
