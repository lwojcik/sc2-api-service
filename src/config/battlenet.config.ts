import { registerAs } from '@nestjs/config';
import { RegionName } from 'blizzapi';

export const battleNetConfig = registerAs('battlenet', () => ({
  region: process.env.SAS_BATTLENET_REGION as RegionName,
  clientId: process.env.SAS_BATTLENET_CLIENT_ID,
  clientSecret: process.env.SAS_BATTLENET_CLIENT_SECRET,
  timeoutMs: parseInt(process.env.SAS_BATTLENET_CLIENT_SECRET, 10),
}));
