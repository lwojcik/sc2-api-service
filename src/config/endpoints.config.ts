import { registerAs } from '@nestjs/config';

export const endpointsConfig = registerAs('endpoints', () => ({
  status: {
    url: '/status',
    method: 'GET',
  },
  getLeague: {
    url: '/data/league/:seasonId/:queueId/:teamType/:leagueId',
    method: 'GET',
  },
  getGrandmaster: {
    url: '/ladder/grandmaster/:regionId',
    method: 'GET',
  },
  getSeason: {
    url: '/ladder/season/:regionId',
    method: 'GET',
  },
  getStatic: {
    url: '/profile/static/:regionId',
    method: 'GET',
  },
  getMetadata: {
    url: '/profile/metadata/:regionId/:realmId/:profileId',
    method: 'GET',
  },
  getProfile: {
    url: '/profile/profile/:regionId/:realmId/:profileId',
    method: 'GET',
  },
  getLadderSummary: {
    url: '/profile/laddersummary/:regionId/:realmId/:profileId',
    method: 'GET',
  },
  getPlayerLadder: {
    url: '/profile/ladder/:regionId/:realmId/:profileId/:ladderId',
    method: 'GET',
  },
  getLegacyProfile: {
    url: '/legacy/profile/:regionId/:realmId/:profileId',
    method: 'GET',
  },
  getLegacyLadders: {
    url: '/legacy/ladders/:regionId/:realmId/:profileId',
    method: 'GET',
  },
  getLegacyMatches: {
    url: '/legacy/matches/:regionId/:realmId/:profileId',
    method: 'GET',
  },
  getLegacyLadder: {
    url: '/legacy/ladder/:ladderId',
    method: 'GET',
  },
  getLegacyAchievements: {
    url: '/achievements/:regionId',
    method: 'GET',
  },
}));
