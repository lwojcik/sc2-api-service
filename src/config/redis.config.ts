import { registerAs } from '@nestjs/config';

export const redisConfig = registerAs('redis', () => ({
  enable: process.env.SAS_REDIS_ENABLE === 'true',
  host: process.env.SAS_REDIS_HOST,
  port: parseInt(process.env.SAS_REDIS_PORT, 10),
  password: process.env.SAS_REDIS_PASSWORD,
  ttlSecs: parseInt(process.env.SAS_REDIS_TTL_SECS, 10),
  db: parseInt(process.env.SAS_REDIS_DB, 10),
  keyPrefix: 'sas',
  expire: process.env.SAS_REDIS_EXPIRE === 'true',
}));
