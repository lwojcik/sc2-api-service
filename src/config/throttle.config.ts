import { registerAs } from '@nestjs/config';

export const throttleConfig = registerAs('throttle', () => ({
  ttlSecs: parseInt(process.env.SAS_THROTTLE_TTL_SECS, 10),
  limit: parseInt(process.env.SAS_THROTTLE_LIMIT, 10),
}));
