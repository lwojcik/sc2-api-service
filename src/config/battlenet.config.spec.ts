import { battleNetConfig } from './battlenet.config';

describe('battleNetConfig', () => {
  it('should create Battle.net config', () => {
    expect(battleNetConfig()).toMatchSnapshot();
  });
});
