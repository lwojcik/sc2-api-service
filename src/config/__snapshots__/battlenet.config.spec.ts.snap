// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`battleNetConfig should create Battle.net config 1`] = `
Object {
  "clientId": undefined,
  "clientSecret": undefined,
  "region": undefined,
  "timeoutMs": 20000,
}
`;
